apps/web/src/hooks/useApproveCallback.ts

```diff
@@ -140,22 +150,47 @@ export function useApproveCallback(
       if (!estimatedGas) return undefined
       const finalAmount =
         overrideAmountApprove ?? (useExact ? amountToApprove?.quotient ?? targetAmount ?? MaxUint256 : MaxUint256)
-      const res = callWithGasPrice(tokenContract, 'approve' as const, [spender as Address, finalAmount], {
-        gas: calculateGasMargin(estimatedGas),
-      })
+
+      let sendTxResult: Promise<SendTransactionReturnType> | undefined
+
+      if (enablePaymaster && isPaymasterAvailable && isPaymasterTokenActive) {
+        const calldata = encodeFunctionData({
+          abi: parseAbi(['function approve(address spender, uint256 amount) public returns (bool)']),
+          functionName: 'approve',
+          args: [spender as Address, finalAmount],
+        })
+
+        const call = {
+          address: tokenContract.address,
+          gas: estimatedGas,
+          value: '0x0' as Hex,
+          calldata,
+        }
+
+        sendTxResult = sendPaymasterTransaction(call, account as Address)
```

Comment:
Please do not use type assertion if possible. If the value is possibly undefined, then let's make sure it's not, or we need to accept undefined for `sendPaymasterTransaction`